### PR TITLE
#21 number.power

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Closes: #21

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@Graur please take a look

File: src/test/eo/org/eolang/math/number-tests.eo [hunk 1]
```diff
@@ -385,6 +385,147 @@
...
+  assert-that > @
+    number 2
+    .power 0
+    $.equal-to 1
+
+[] > power-is-float-zero-of-int
+  assert-that > @
+    number 2
+    .power 0.0
+    $.equal-to 1
```
Comment (Graur) [inline src/test/eo/org/eolang/math/number-tests.eo:437]: @maxonfjvipon Can you explain, why we return `int` here? In `float-power-of-int-zero` test we return `float` in such case.

File: src/test/eo/org/eolang/math/number-tests.eo [hunk 2]
```diff
@@ -385,6 +385,147 @@
...
+
+[] > power-is-int-zero-of-float
+  assert-that > @
+    number 2.2
+    .power 0
+    $.equal-to 1.0
```
Comment (Graur) [inline src/test/eo/org/eolang/math/number-tests.eo:443]: @maxonfjvipon Same here, can you explain, why we return `float` here?

**Review by Graur — CHANGES_REQUESTED** [review]
@maxonfjvipon Can you clarify a couple of questions?

File: src/test/eo/org/eolang/math/number-tests.eo (hunk 1 above)
Comment (maxonfjvipon) [reply src/test/eo/org/eolang/math/number-tests.eo:437]: You are right. It should be float here

File: src/test/eo/org/eolang/math/number-tests.eo (hunk 2 above)
Comment (maxonfjvipon) [reply src/test/eo/org/eolang/math/number-tests.eo:443]: My idea was: if both arguments are int - return int, if at least one is float - return float. So I think there is no mistake here

Comment (Graur) [reply src/test/eo/org/eolang/math/number-tests.eo:443]: @maxonfjvipon Now I get it, thanks!

File: src/test/eo/org/eolang/math/number-tests.eo (hunk 1 above)
Comment (maxonfjvipon) [reply src/test/eo/org/eolang/math/number-tests.eo:437]: @Graur it seems `eo` repository has not included [this](https://github.com/objectionary/eo/issues/942) to the release.  
 So this test won't work with `$.equal-to 1.0` until it's done

Comment (Graur) [reply src/test/eo/org/eolang/math/number-tests.eo:437]: @maxonfjvipon I believe it was released

Comment (maxonfjvipon) [reply src/test/eo/org/eolang/math/number-tests.eo:437]: @Graur but why it fails [here](https://github.com/objectionary/eo-math/pull/41) then?

Comment (Graur) [reply src/test/eo/org/eolang/math/number-tests.eo]: @maxonfjvipon let me check it

Comment (maxonfjvipon) [reply src/test/eo/org/eolang/math/number-tests.eo]: @Graur I ran it locally
image

Comment (Graur) [reply src/test/eo/org/eolang/math/number-tests.eo]: @maxonfjvipon Indeed, you are right! I can't see these changes in 0.27.1 version of EO

Comment (maxonfjvipon) [reply src/test/eo/org/eolang/math/number-tests.eo]: @Graur could you please check if 0.27.2 version of EO contains needed changes?

Comment (Graur) [reply src/test/eo/org/eolang/math/number-tests.eo]: @maxonfjvipon You can check it yourself by simply investigating the "target" folder. I believe it should be there

**Comment by maxonfjvipon** [comment]
@Graur please take a look

**Review by Graur — APPROVED** [review]
@yegor256 please, have a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**